cdap-api/src/main/java/co/cask/cdap/api/app/AbstractApplication.java

```diff
@@ -225,24 +227,25 @@ protected void addService(String name, HttpServiceHandler handler, HttpServiceHa
    * @param schedule the schedule to be added for the Workflow
    * @param workflowName the name of the Workflow
    */
+  @Deprecated
```

Comment:
I think it makes sense that they both extend a ``Schedule`` class, since, well, they are schedules... Also because the ``Scheduler`` interface works on ``Schedule`` objects. I wouldn't see the point of copying its methods to handle ``StreamSizeSchedule`` objects specifically.